agent/tool-scripts/postprocess/pidstat-convert

```diff
@@ -0,0 +1,114 @@
+my %current_stats;
+my @attributes = qw(usr system guest CPU_PCT CPUID minflt_s majflt_s VSZ RSS MEM_PCT kB_rd_s kB_wr_s iodelay cswch_s nvcswch_s);
+my $tool_output_dir = shift; # the one and only argument to this script, the directory where these file shoudl be written
+if (not defined $tool_output_dir) {
+	print "Directory to store PIDs is required\n";
+	exit 1;
+}
+my %file_handles;
+if ( -e $tool_output_dir . '/pids') {
+	rmtree($tool_output_dir . '/pids');
+}
+mkdir($tool_output_dir . '/pids');
+# read the STDIN which is expcted to be STDOUT from 'pidstat -l -H -w -u -h -d -r  -p ALL'
+my $prev_pid;
+while (my $line = <>) {
+	# example STDIN
+	#
+	###v11 no threads
+        ## Time        UID       PID    %usr %system  %guest   %CPU   CPU  minflt/s  majflt/s     VSZ     RSS   %MEM   kB_rd/s   kB_wr/s kB_ccwr/s iodelay   cswch/s nvcswch/s  Command
+        # 1547780606     0      2368    0.00    0.00    0.00    0.00     4      0.00      0.00  112808    4300   0.00      0.00      0.00      0.00       0      0.00      0.00  /usr/sbin/sshd -D
+        # 1547780606     0      2373    0.00    0.00    0.00    0.00     0      0.00      0.00  228772    7940   0.00      0.00      0.00      0.00       0      0.00      0.00  /usr/sbin/rsyslogd -n
+        # 1547780606     0      2380    0.00    0.00    0.00    0.00     7      0.00      0.00 1221772   15360   0.01      0.00      0.00      0.00       0      0.00      0.00  /usr/sbin/libvirtd
+        
+	###v11 with threads
+        # Time   UID      TGID       TID    %usr %system  %guest    %CPU   CPU  minflt/s  majflt/s     VSZ     RSS   %MEM   kB_rd/s   kB_wr/s kB_ccwr/s iodelay   cswch/s nvcswch/s  Command
+        # 1547852254     0         1         0    0.00    0.00    0.00    0.00     4      0.13      0.00  194520    7572   0.00      1.16     28.14      1.62      89      0.62      0.01  /usr/lib/systemd/systemd --switched-root --system --deserialize 22
+        # 1547852254     0         0         1    0.00    0.00    0.00    0.00     4      0.13      0.00  194520    7572   0.00      0.02      0.00      0.00      89      0.62      0.01  |__/usr/lib/systemd/systemd --switched-root --system --deserialize 22
+        #
+        if ( $line =~ /\s{0,1}(\d+)\s+\d+\s+(\S+|\S+\s+\S+)\s+(\d+\.\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\s+\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\.\d+\s+\d+\s+\d+\.\d+\s+\d+\.\d+\s+)(.*)/ ) {
+		my $time = $1;
+		my $pid = $2;
+		my $values_str = $3;
+		my $cmd = substr $4, 0, 40; # limit the length to 60 chars
+		my @values = split(/\s+/, $values_str);
+		if ($pid =~ /(^\S+)\s+(\S+)$/) {
+			my $tgid = $1;
+			my $tid = $2;
+			if ($tgid =~ /^0|\-$/) {
+				$pid = $prev_pid . "-" . $tid;
+				$cmd =~ s/\|__//;
+			} else {
+				$prev_pid = $tgid;
+				next; # Not going to save the per-process stats because we have the per-thread stats
+			}
+		}
+		# filenames have restrictions, and a process name with arguments don't always work as filename
```

Comment:
"..,a process name with arguments does not always work as a filename."

Reply:
Fix in #1046 PR.